### ISSUE TITLE
Play 2.9 incompatible with akka-http 10.6.x when using https

### ISSUE THREAD

**Issue by dwickern** [playframework/playframework#12711]
### Play Version

2.9.3

### Library Dependencies

akka-http 10.6.3
akka 2.9.3
scala 2.13.14

### Actual Behavior

1. Configure play server with https, e.g. by adding `https.port = 443` to `application.conf`
2. Start the server and the following error will occur:
```
java.lang.NoSuchMethodError: 'scala.Option akka.http.scaladsl.ConnectionContext$.https$default$2()'
	at play.core.server.AkkaHttpServer.$anonfun$httpsServerBinding$1(AkkaHttpServer.scala:259)
	at play.core.server.AkkaHttpServer.$anonfun$httpsServerBinding$1$adapted(AkkaHttpServer.scala:255)
	at scala.Option.map(Option.scala:242)
	at play.core.server.AkkaHttpServer.<init>(AkkaHttpServer.scala:255)
	at play.core.server.AkkaHttpServerProvider.createServer(AkkaHttpServer.scala:742)
	at play.core.server.AkkaHttpServerProvider.createServer(AkkaHttpServer.scala:740)
	at play.core.server.ServerProvider.createServer(ServerProvider.scala:30)
	at play.core.server.ServerProvider.createServer$(ServerProvider.scala:27)
	at play.core.server.AkkaHttpServerProvider.createServer(AkkaHttpServer.scala:740)
	at play.core.server.ProdServerStart$.start(ProdServerStart.scala:58)
	at play.core.server.ProdServerStart$.main(ProdServerStart.scala:28)
	at play.core.server.ProdServerStart.main(ProdServerStart.scala)
Oops, cannot start the server.
```

Timeline:
- In akka-http 10.2.0, `ConnectionContext.https` was deprecated in favor of `ConnectionContext.httpsServer` or `ConnectionContext.httpsClient`. The APIs are different so it's not a simple rename.
- In akka-http 10.6.0, `ConnectionContext.https` was removed.

**Comment by dwickern** [comment]
The most up-to-date working configuration seems to be
- play 2.9.3
- akka-http 10.5.3
- akka 2.8.5 (because akka-http 10.5.3 is not compatible with akka-streams 2.9.x)

**Comment by mkurz** [comment]
@dwickern so you know what's going on, can you open a PR against the 2.9.x branch with a fix?

**Comment by mkurz** [comment]
- Fixed by #12719
